driver/random.hpp

```diff
@@ -2,15 +2,18 @@
 #define GUARD_RANDOM_GEN_
 
 #include <cstdlib>
+#include <ctime>
+#include <cstdint>
 
 template <typename T>
 inline T FRAND()
 {
-    double d = static_cast<double>(rand() / (static_cast<double>(RAND_MAX)));
+    std::uint32_t seed = time(nullptr);
+    double d = static_cast<double>(rand_r(&seed) / (static_cast<double>(RAND_MAX)));
```

Comment:
And I would not recommend using `rand_r()` since it's kind of non-standard function.
It is a part of POSIX, not C\C++ standard and formally speaking it is deprecated.

Have you considered to use `std::generate_canonical` for [0;1) fp distribution?

Reply:
I am inclining towards the standards side for sure. What's the reasoning behind `std::generate_canonical` though? Why a fp in [0;1) specifically?